### DISCUSSION TITLE
[v5] Refresh token

### DISCUSSION THREAD

**Discussion by victor-sc-92** [adonisjs/core#2580] [Help] [answered]
How can i use refresh token in v5?
Is there any forecast to be implemented?

**Comment by RomainLanz** [comment, marked as answer]
Hey @CodeAkio! 👋 

We do not plan to add JWT into the core of the auth module for the moment.
Feel free to use any standard JWT library if you want to have JWT Auth inside your application.

**Reply by victor-sc-92** [reply]
But is it possible to do something like that using the api token?
https://docs.adonisjs.com/guides/auth/api-tokens-guard

Because i want that the user does not need to pass the credentials (email, password) again after the token expires.

**Reply by RomainLanz** [reply]
You can increase the expiration time of the API Token to avoid your user to be logged out.

**Reply by victor-sc-92** [reply]
But isn't that a security issue?

**Reply by RomainLanz** [reply]
Why would that be a security issue?

JWT grants a false feeling of security. Your Refresh Token also has a vast (or none) expiration date if you think about it.

**Reply by victor-sc-92** [reply]
I may be wrong, but u think that, the refresh token allows you to generate a new token when the later one expires.
So I can have a token of 15 min and a refresh of 1 month, the token always changes, so if the token is compromised, the attacker will only have 15 min of access and then lose.

When extending the token expiration with each access, as I understand it, if the token is compromised, the attacker will have unlimited access.

At least that's how I understood it.

**Reply by thetutlage** [reply]
If the JWT token is compromised, what will stop the refresh token from getting compromised? And if the attacker has the refresh token, they can generate as many JWT tokens as they want.

**Reply by RomainLanz** [reply]
Alright, what happen if your refresh token leaks?

**Reply by victor-sc-92** [reply]
It is true 🤣 , thank you very much for the clarification and patience. 👍
